shared/selene/util/log.py

```diff
@@ -40,61 +40,60 @@
-    """Configure a logger with a daily log file and a console log"""
-    def __init__(self, logger_name):
-        self.logger = getLogger()
-        self.logger.level = DEBUG
-        self.file_log_level = DEBUG
-        self.console_log_level = INFO
-        self.log_file_path = path.join('/var/log/mycroft', logger_name + '.log')
-        self.log_msg_formatter = Formatter(
-            '{asctime} | {levelname:8} | {process:5} | {name} | {message}',
-            style='{'
-        )
+def _generate_log_config(service: str) -> dict:
+    """Uses Python's dictionary config for logging to setup Selene logs.
 
-    def _define_file_handler(self):
-        """build a file handler"""
-        handler = handlers.TimedRotatingFileHandler(
-            filename=self.log_file_path,
-            when='midnight',
-            utc=True
-        )
-        handler.setLevel(self.file_log_level)
-        handler.setFormatter(self.log_msg_formatter)
+    Args:
+        service: the name of the service initiating the log setup
 
-        return handler
+    Returns:
+        The logging configuration in dictionary format.
+    """
+    log_format = (
+        "{asctime} | {levelname:8} | {process:5} | {name}.{funcName} | {message}"
+    )
+    default_formatter = {"format": log_format, "style": "{"}
+    console_handler = {
+        "class": "logging.StreamHandler",
+        "formatter": "default",
+        "stream": "ext://sys.stdout",
+    }
+    file_handler = {
+        "class": "logging.handlers.TimedRotatingFileHandler",
+        "formatter": "default",
+        "filename": f"/var/log/mycroft/{service}.log",
+        "backupCount": 30,
+        "when": "midnight",
+    }
 
-    def _define_console_handler(self):
-        """build a console, or stream, handler"""
-        handler = StreamHandler()
-        handler.setLevel(self.console_log_level)
-        handler.setFormatter(self.log_msg_formatter)
+    return {
+        "version": 1,
+        "formatters": {"default": default_formatter},
+        "handlers": {"console": console_handler, "file": file_handler},
+        "root": {"level": "INFO", "handlers": ["file"]},
+    }
 
-        return handler
 
-    def configure(self):
-        """Put it all together"""
-        file_handler = self._define_file_handler()
-        console_handler = self._define_console_handler()
-        self.logger.addHandler(file_handler)
-        self.logger.addHandler(console_handler)
+def configure_selene_logger(service):
+    """Configures the base logger for any Selene service or application.
 
+    Args:
+        service: the name of the service initiating the log setup
+    """
+    log_level = environ.get("SELENE_LOG_LEVEL", "INFO")
+    log_config = _generate_log_config(service)
+    selene_logger = {
+        "selene": {"level": log_level, "handlers": ["console", "file"], "propagate": 0}
+    }
+    log_config["loggers"] = selene_logger
+    logging.config.dictConfig(log_config)
+    logging.getLogger("selene")
```

Comment:
This did confuse me a little in working out what was part of the base selene logger and what was being generated for each of the (child) service loggers.

Does the `selene_logger` need to be included each time or it just is because it doesn't hurt to?
Is there a reason why the root config is defined in the generate function but the selene logger is defined here? 

I don't think any of this makes a difference in practice though so feel free to ignore.

Reply:
There is a difference between the library code and the services.  The library code is in a "selene" package so when you specify `__name__` in the `logging.getLogger()` function, it automatically becomes a child of the selene logger.  The services are in packages named "public_api" or "account_api".  So they need to call `get_selene_logger(__name__)` to ensure the highest level in the logging hierarchy is "selene".